File: src/main/scala/resonant/content/wrapper/RenderTileDummy.scala
package resonant.content.wrapper

import resonant.content.spatial.block.SpatialBlock
import resonant.content.component.render.RenderDynamic
import net.minecraft.tileentity.TileEntity
import universalelectricity.core.transform.vector.Vector3

/**
 * @author Calclavia
 */
@SideOnly(Side.CLIENT)
class RenderTileDummy extends TileEntitySpecialRenderer
{
  def renderTileEntityAt(tile: TileEntity, x: Double, y: Double, z: Double, f: Float)
  {
    if (tile.isInstanceOf[SpatialBlock])
    {
      val spatial = tile.asInstanceOf[SpatialBlock]
      spatial.components.get(classOf[RenderDynamic]).foreach(_.renderDynamic(new Vector3(x, y, z), false, f))
    }
  }
}

File: src/main/scala/com/builtbroken/mc/lib/render/block/RenderTileDummy.java
package com.builtbroken.mc.lib.render.block;

import com.builtbroken.mc.imp.transform.vector.Pos;
import com.builtbroken.mc.prefab.tile.Tile;
import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
import net.minecraft.tileentity.TileEntity;

/**
 * Created by robert on 1/11/2015.
 */
public class RenderTileDummy extends TileEntitySpecialRenderer
{
    @Override
    public void renderTileEntityAt(TileEntity tile, double x, double y, double z, float f)
    {
        if(tile instanceof Tile)
        {
            ((Tile) tile).renderDynamic(new Pos(x, y, z), f, 0);
        }
    }
}
